library/core/src/iter/adapters/intersperse.rs

```diff
@@ -151,17 +151,15 @@ where
 {
     let (lo, hi) = iter.size_hint();
     let next_is_elem = !needs_sep;
-    let lo = lo.saturating_sub(next_is_elem as usize).saturating_add(lo);
-    let hi = match hi {
-        Some(hi) => hi.saturating_sub(next_is_elem as usize).checked_add(hi),
-        None => None,
-    };
-    (lo, hi)
+    (
+        lo.saturating_sub(next_is_elem as usize).saturating_add(lo),
+        hi.and_then(|hi| hi.saturating_sub(next_is_elem as usize).checked_add(hi)),
+    )
 }
 
 fn intersperse_fold<I, B, F, G>(
     mut iter: Peekable<I>,
-    init: B,
+    mut accum: B,
```

Comment:
I think the previous field name here might've been better, as it's consistent with `init` used in all other `Iterator` combinators.

Reply:
Force-pushed it away :-)